File: matlab/ex1/computeCost.m
function J = computeCost( X,Y,theta )
%% 函数功能：计算误差平方和作为代价值
%  X：增加了一个维度后的输入特征
%  J：X对应的输出值
%  theta：带入的参数
[~,m]=size(X);
hypthesis = theta.' * X;
err = hypthesis - Y;
J = (err * err.') / (2*m);
end

File: python/ex1/computeCost.py
"""计算代价值"""
import numpy as np
def compute_cost(X,Y,theta):
	# 两个数组作矩阵乘积
	# 当两个数组的维度不能直接进行矩阵乘法时，dot会把后面的参数进行转置
	hypthesis = np.dot(X,np.transpose(theta))
	# 先转置再做矩阵乘法
	cost = np.dot(np.transpose(hypthesis - Y),(hypthesis -Y))
	cost = cost / (2 * X.shape[0])
	return cost